packages/block-library/src/media-text/media-container.native.js

```diff
@@ -168,116 +169,135 @@ class MediaContainer extends Component {
 
 	renderImage( params, openMediaOptions ) {
 		const { isUploadInProgress } = this.state;
-		const { mediaAlt, mediaUrl, isSelected } = this.props;
 		const {
-			finalWidth,
-			finalHeight,
-			imageWidthWithinContainer,
-			isUploadFailed,
-			retryMessage,
-		} = params;
-		const opacity = isUploadInProgress ? 0.3 : 1;
-
-		const contentStyle = ! imageWidthWithinContainer
-			? styles.content
-			: styles.contentCentered;
+			aligmentStyles,
+			focalPoint,
+			imageFill,
+			isMediaSelected,
+			isSelected,
+			mediaAlt,
+			mediaUrl,
+			mediaWidth,
+			shouldStack,
+		} = this.props;
+		const { isUploadFailed, retryMessage } = params;
+		const focalPointValues =
+			imageFill && ! focalPoint ? { x: 0.5, y: 0.5 } : focalPoint;
 
 		return (
-			<TouchableWithoutFeedback
-				accessible={ ! isSelected }
-				onPress={ this.onMediaPressed }
-				onLongPress={ openMediaOptions }
-				disabled={ ! isSelected }
+			<View
+				style={ [
+					imageFill && styles.imageWithFocalPoint,
+					imageFill &&
+						shouldStack && {
+							height: styles.imageFill.height,
+						},
+				] }
 			>
-				<View style={ contentStyle }>
-					{ ! imageWidthWithinContainer && (
-						<View style={ styles.imageContainer }>
-							{ this.getIcon( false ) }
-						</View>
-					) }
-					<ImageBackground
-						accessible={ true }
-						accessibilityLabel={ mediaAlt }
-						accessibilityHint={ __(
-							'Double tap and hold to edit'
-						) }
-						accessibilityRole={ 'imagebutton' }
-						style={ {
-							width: finalWidth,
-							height: finalHeight,
-							opacity,
-						} }
-						resizeMethod="scale"
-						source={ { uri: mediaUrl } }
-						key={ mediaUrl }
+				<TouchableWithoutFeedback
+					accessible={ ! isSelected }
+					onPress={ this.onMediaPressed }
+					onLongPress={ openMediaOptions }
+					disabled={ ! isSelected }
+				>
+					<View
+						style={ [
+							styles.mediaImageContainer,
+							! isUploadInProgress && aligmentStyles,
+						] }
 					>
-						{ isUploadFailed && (
-							<View
-								style={ [
-									styles.imageContainer,
-									styles.uploadFailed,
-								] }
-							>
-								<View style={ styles.modalIcon }>
-									{ this.getIcon( isUploadFailed ) }
-								</View>
-								<Text style={ styles.uploadFailedText }>
-									{ retryMessage }
-								</Text>
-							</View>
-						) }
-					</ImageBackground>
-				</View>
-			</TouchableWithoutFeedback>
+						<Image
+							align="center"
+							alt={ mediaAlt }
+							focalPoint={ imageFill && focalPointValues }
```

Comment:
Hmmm i feel like the `imageFill` check is duplicated here. 

Maybe it could be like:
```
const focalPointValues = ! focalPoint ? { x: 0.5, y: 0.5 } : focalPoint;
```
and then we could leave the check only here? Am i right or i miss something?

Reply:
Yup, it was! Good catch! I changed it as you suggested. Thanks!